keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1408 @@
+recently [became more
+acceptable](https://github.com/kubernetes/enhancements/pull/2537). Kube-scheduler
+and plugin communicate by modifying that status. The status is also
+how Kubernetes tracks that a plugin has allocated the resource and on
+which nodes the resource is available.
+
+This approach is an intentional simplification compared to the PV/PVC
+model for volumes because we don't need to deal with two objects when
+allocating resources and therefore don't need something like the
+volume binding controller. If desired, a resource plugin can implement
+support for manually allocated (“static provisioning” in the context
+of volumes) and/or existing resources by reacting to ResourceClaims by
+using those resources to satisfy a claim.
+
+Allocation of a resource happens either immediately (“immediate
+allocation”) or when a Pod needs the resource (“delayed allocation”),
+depending on a flag in the ResourceClaim spec. Pods reference resource
+claims in a new PodSpec.Resources list. Each resource in that list
+then can be made available to one or more containers in that Pod.  To
+support ephemeral resources, an entry in the new PodSpec.Resources
+list can also be a ResourceClaimTemplate. When a Pod gets created,
+such a template will be used to create a normal ResourceClaim with the
+Pod as owner, and then the normal allocation of the resource takes
+place.
+
+For immediate allocation, scheduling Pods is simple because the
+resource is already allocated and determines the nodes on which the
+Pod may run. For delayed allocation, a node is selected tentatively
+and plugin(s) try to allocate their resources for that node. If that
+succeeds, the Pod can start to run. If it fails, the scheduler must
+determine whether some other node fits the requirements and if so,
+request allocation again. If no node fits because some resources were
+already allocated for a node and are only usable there, then those
+resources must be released and then get allocated elsewhere.
+
+The resources allocated for a ResourceClaim can be shared by multiple
+containers in a pod. Depending on the capabilities defined in the
+ResourceClaim by the plugin, a ResourceClaim can be used exclusively
+by one pod at a time, by a certain maximum number of pods, or an
+unlimited number of pods.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Partial allocation.
+
+Pod requests GPU resource with 2Gb memory:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: device-consumer
+spec:
+  containers:
+  - name: my-ubuntu
+    image: ubuntu
+    command: ["/bin/program"]
+    resources:
+    - name: gpu_2Gb
+  resources:
+  - name: gpu_2Gb
+    template:
+      resourceClassName: "gpu"
+      parameters:
+        memory: "2Gb"
```

Comment:
To be clear, it wouldn't look exactly like this, right? You still need to support the direct fields of the "traditional" `resources` key as well (i.e. request / limits and any subkeys below that, i.e. cpus, memory, (any custom resources), etc.).

Reply:
"resources" here is the new field at the PodSpec level. In the container, the example incorrectly also used "resources" (the existing, stable field name) instead of the new one.